apps/opik-frontend/src/lib/rules.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  EvaluatorsRule,
+  EVALUATORS_RULE_TYPE,
+  LLMJudgeDetails,
+  PythonCodeDetails,
+} from "@/types/automations";
+import { LLMJudgeSchema } from "@/types/llm";
+
+export const isPythonCodeRule = (rule: EvaluatorsRule): boolean => {
+  return (
+    rule.type === EVALUATORS_RULE_TYPE.python_code ||
+    rule.type === EVALUATORS_RULE_TYPE.thread_python_code ||
+    rule.type === EVALUATORS_RULE_TYPE.span_python_code
+  );
+};
+
+/**
+ * Attempts to extract metric name from Python code by parsing the __init__ default parameter.
+ * Looks for patterns like: def __init__(self, name: str = "my_custom_metric")
+ * Returns empty array if no name can be extracted.
+ */
+const extractMetricNameFromPythonCode = (code: string): string[] => {
```

Comment:
This solution looks scary, and depends on Python syntax, and Python version

Isn't there another way? With regex, feels somewhat like a reflection/hacking

There should be alternatives, whether tree-sitter/filbert, Python Lexer, user providing a small input, or have BE parse with **ast**

Reply:
Feature does work, as checked in testing-environment https://pr-4528.dev.comet.com from recent version

Suggesting to consider alternatives in a tech debt ticket, so it doesn't block the PR

Created ticket:
https://comet-ml.atlassian.net/browse/OPIK-3629

FYI @aadereiko